challenge.rb

```diff
@@ -1,26 +1,26 @@
 
 def capitalize_each_string(input)
-  #implement your solution here
+	input.map {|word| word.capitalize}
 end
 
 def fetch_the_dog(input)
-  #implement your solution here
+	input.select {|word| word=="dog"}
 end
 
 def no_dogs_allowed(input)
-  #implement your solution here
+	input.reject {|word| word=="dog"}
 end
 
 def count_the_animals(input)
-  #implement your solution here
+	input.count
 end
 
 def fetch_the_first_two(input)
-  #implement your solution here
+	input.first(2)
 end
 
 def fetch_CD_animals(input)
-  #implement your solution here
+	input.keep_if{|word| word[0]=="c"|| word[0]=="d"}
```

Comment:
This clearly says "keep words when the first character is `c` or the first character is `d`". 👍 

You'll find over time that Ruby has many builtin methods to make little things like this even more clear. Check out [String #start_with?](http://ruby-doc.org/core-2.0.0/String.html#method-i-start_with-3F).